stockroom/external/importer/torchvision_importers.py

```diff
@@ -41,26 +41,61 @@ def variability_status(self):
     def __len__(self):
         return len(self.dataset)
 
+    @classmethod
+    def gen_splits(cls, torchvision_dataset, root):
+        dataset_splits = []
+        for train in [True, False]:
+            dataset = torchvision_dataset(root=root, train=train, download=True)
+            dataset_splits.append(cls(dataset, train))
+        return dataset_splits
+
 
 class Cifar10(TorchvisionCommon):
     name = 'cifar10'
 
-    def __init__(self, root, train=True):
-        dataset = datasets.CIFAR10(root=root, train=train, download=True)
-        super().__init__(dataset, train)
+    @classmethod
+    def gen_splits(cls, root):
+        dataset = datasets.CIFAR10
+        dataset_splits = super().gen_splits(dataset, root)
+        return dataset_splits
+
+    @staticmethod
+    def _process_data(img, lbl):
+        img = np.transpose(np.array(img), (2, 0, 1))
+        img = np.ascontiguousarray(np.array(img))
+        img = img.astype(np.float32) / 255
+        lbl = np.array(lbl)
+        return img, lbl
 
 
 class Mnist(TorchvisionCommon):
     name = 'mnist'
 
-    def __init__(self, root, train=True):
-        dataset = datasets.MNIST(root=root, train=train, download=True)
-        super().__init__(dataset, train)
+    @classmethod
+    def gen_splits(cls, root):
+        dataset = datasets.MNIST
+        dataset_splits = super().gen_splits(dataset, root)
+        return dataset_splits
 
 
 class FashionMnist(TorchvisionCommon):
     name = 'fashion_mnist'
 
-    def __init__(self, root, train=True):
-        dataset = datasets.FashionMNIST(root=root, train=train, download=True)
-        super().__init__(dataset, train)
+    @classmethod
+    def gen_splits(cls, root):
+        dataset = datasets.FashionMNIST
+        dataset_splits = super().gen_splits(dataset, root)
+        return dataset_splits
+
```

Comment:
`FashionMnist` also need to override the `_process_data` from the base class, right?

Reply:
is it required? Since process is a static method its already in the namespace when we inherit rt?
I'm not sure since I not clear about how inheritance of static method works but the code is calling the `_process_data`  of the parent and I wrote the parent `_process_data` to handle mnist and fashionMnist

Reply:
I am sorry. You are right. I was in an assumption that `FashionMnist` has color channels.

Reply:
Oh yeah got it. My bad :sweat_smile: 
fixed it!